File: src/main/java/com/java/scogen/dataGenerator/EmployeeNameList.java
package com.java.scogen.dataGenerator;

import java.util.HashSet;
import java.util.Random;
import java.util.Set;

public class EmployeeNameList {
    static int counter = 0;

    public Set<String> setofUniqueNames() {
        Set<String> nameSet = new HashSet<>();
        while (nameSet.size() < 1_000)
        {
            nameSet.add(randomNameGenerator());
        }

        return nameSet;
    }

    public static String randomNameGenerator() {
        int leftLimit = 48;
        int rightLimit = 122;
        Random random = new Random();
        int targetStringLength = random.nextInt(10)+3;

        return random.ints(leftLimit, rightLimit + 1)
                .filter(i -> (i <= 57 || i >= 65) && (i <= 90 || i >= 97))
                .limit(targetStringLength)
                .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
                .toString();
    }

    public static String randomIDGenerator() {
        return Integer.toString(counter++);
    }

    public static int randomAgeGenerator() {
        int leftLimit = 18;
        int rightLimit = 42;
        Random random = new Random();

        return random.nextInt(rightLimit + 1)+leftLimit;
    }

}

File: src/main/scala/com/scala/scogen/metaData/EmployeeNameList.scala
package com.scala.scogen.metaData

import scala.collection.immutable.HashSet
import scala.util.Random

object EmployeeNameList {

  private[metaData] var counter = 0

  def setofUniqueNames: Set[String] = {
    var nameSet = new HashSet[String]
    while (nameSet.size < 1000)
      {
        nameSet += randomNameGenerator
      }
    nameSet
  }

  def randomNameGenerator: String = {
    val alpha = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
    val size = alpha.size
    val random = new Random
    val targetStringLength = random.nextInt(10) + 3
    (1 to targetStringLength).map(x => alpha(random.nextInt().abs % size)).mkString
  }

  def serialIDGenerator: String = Integer.toString({
    counter += 1; counter - 1
  })

  def randomAgeGenerator: Int = {
    val leftLimit = 18
    val rightLimit = 42
    val random = new Random
    random.nextInt(rightLimit + 1) + leftLimit
  }

}
